temporian/implementation/numpy/data/event.py

```diff
@@ -89,6 +105,165 @@ def schema(self) -> Event:
             sampling=self.sampling.names,
         )
 
+    @staticmethod
+    def from_dataframe(
+        df: pd.DataFrame,
+        index_names: List[str],
+        timestamp_column: str = "timestamp",
+    ) -> "NumpyEvent":
+        """Convert a pandas DataFrame to a NumpyEvent. Supported
+        dtypes are: np.float64, np.float32, np.int64, np.int32, np.datetime64
+        (for timestamp column).
+
+        Args:
+            df: DataFrame to convert to NumpyEvent.
+            index_names: names of the DataFrame columns to be used as index for the event.
+            timestamp_column: Name for timestamp index. Defaults to "timestamp".
```

Comment:
Column containing timestamps. Supported types: {...}. Timestamps of the types [...] are converted implicitly to in UTC epoch.